packages/nimble-components/src/rich-text/models/tests/markdown-serializer.spec.ts

```diff
@@ -1,415 +1,306 @@
-            },
-            {
-                name: 'HTML entities <&euro>',
-                html: '&euro;',
-                markdown: '€'
-            },
-            {
-                name: 'Markdown syntax strings <*>',
-                html: '*',
-                markdown: String.raw`\*`
-            },
-            {
-                name: 'Markdown syntax strings <**>',
-                html: '**',
-                markdown: String.raw`\*\*`
-            },
-            {
-                name: 'Markdown syntax strings <_>',
-                html: '_',
-                markdown: String.raw`\_`
-            }
-        ];
-
-        const focused: string[] = [];
-        const disabled: string[] = [];
-        for (const value of supportedNodesMarks) {
-            const specType = getSpecTypeByNamedList(value, focused, disabled);
-            specType(
-                `Should return ${value.name} markdown (${value.markdown}) when its respective node is passed`,
-                // eslint-disable-next-line @typescript-eslint/no-loop-func
-                () => {
-                    const node = getNode(value.html);
-
-                    expect(
-                        RichTextMarkdownSerializer.serializeDOMToMarkdown(node)
-                    ).toBe(value.markdown);
-                }
+    describe('Serialize rich text editor content to its respective markdown', () => {
+        const r = String.raw;
+
+        it('Bold', async () => {
+            await pageObject.clickFooterButton(ToolbarButton.bold);
+            await pageObject.setEditorTextContent('Bold');
+
+            expect(element.getMarkdown()).toEqual('**Bold**');
+        });
+
+        it('Italics', async () => {
+            await pageObject.clickFooterButton(ToolbarButton.italics);
+            await pageObject.setEditorTextContent('Italics');
+
+            expect(element.getMarkdown()).toEqual('*Italics*');
+        });
+
+        it('Link', async () => {
+            await pageObject.setEditorTextContent('https://nimble.ni.dev ');
+            expect(element.getMarkdown()).toEqual('<https://nimble.ni.dev> ');
+        });
+
+        it('Bold and Italics', async () => {
+            await pageObject.clickFooterButton(ToolbarButton.bold);
+            await pageObject.clickFooterButton(ToolbarButton.italics);
+            await pageObject.setEditorTextContent('Bold and Italics');
+            expect(element.getMarkdown()).toEqual('***Bold and Italics***');
+        });
+
+        it('Italics without spaces in between bold texts', async () => {
+            await pageObject.clickFooterButton(ToolbarButton.bold);
+            await pageObject.setEditorTextContent('Bold');
+            await pageObject.clickFooterButton(ToolbarButton.italics);
+            await pageObject.setEditorTextContent('italics');
+            await pageObject.clickFooterButton(ToolbarButton.italics);
+            await pageObject.clickFooterButton(ToolbarButton.italics);
```

Comment:
Why is the test pressing the italics button twice here? Shouldn't that result in it remaining toggled on, so the final segment would also be both bold and italics?

Same for tests below.

Reply:
This is working in some weird way, while looking into it the state toggling of the formatting buttons is managed by two methods, firstly on interaction with toolbar directly and the second way, the button state will be automatically toggled based on the cursor placement in the editor as the function `updateEditorButtonsState()` is binded to every transaction of the editor. So the second way is making these behaviors here.

> await pageObject.clickFooterButton(ToolbarButton.italics);

The above code will toggle the italics to active state.

> await pageObject.setEditorTextContent('italics');

Now adding text contents will be in italics and here the transaction event been triggering `updateEditorButtonsState()` method which making the italics to off state again. I am predicting it is happening because in the test case environment the cursor is not updated properly once after text content been appended. So, to make it work as real flow, I just called it twice in both cases to make it deactivates the formatting.

Reply:
If going that way other toggle apis in the browser are represented as toggle(force?:boolean) so I'd vote for `toggleFooterButton(button: ToolbarButton, force?: boolean)`

Reply:
Added `toggleFooterButton()` method with the `force parameter` and updated the description as well.